12-offer-encoding.md

```diff
@@ -229,6 +229,97 @@ The human-readable prefix for offers is `lno`.
     1. type: 22 (`offer_issuer_id`)
     2. data:
         * [`point`:`id`]
+    1. type: 24 (`offer_recurrence_compulsory`)
+    2. data:
+        * [`recurrence`:`recurrence`]
+    1. type: 25 (`offer_recurrence_optional`)
+    2. data:
+        * [`recurrence`:`recurrence`]
+    1. type: 26 (`offer_recurrence_base`)
+    2. data:
+        * [`recurrence_base`:`recurrence_base`]
+    1. type: 27 (`offer_recurrence_paywindow`)
+    2. data:
+        * [`recurrence_paywindow`:`recurrence_paywindow`]
+    1. type: 29 (`offer_recurrence_limit`)
+    2. data:
+        * [`tu32`:`max_period`]
+
+1. subtype: `recurrence`
+2. data:
+    * [`byte`:`time_unit`]
+    * [`tu32`:`period`]
+
+1. subtype: `recurrence_base`
+2. data:
+    * [`byte`:`proportional_amount`]
+    * [`tu64`:`basetime`]
+
+1. subtype: `recurrence_paywindow`
+2. data:
+    * [`u32`:`seconds_before`]
+    * [`tu32`:`seconds_after`]
+
+### Recurrence
+
+Some offers are *periodic*, such as a subscription service or monthly dues, in that payment is expected to be repeated.  There are many different flavors of repetition, consider:
+
+* Payments due on the first of every month, for 6 months.
+* Payments due on every Monday, 1pm Pacific Standard Time.
+* Payments due once a year:
+   * which must be made on January 1st, or
+   * which are only valid if started January 1st 2021, or
+   * which if paid after January 1st you (over) pay the full rate first year, or
+   * which if paid after January 1st are paid pro-rata for the first year, or
+   * which repeat from whenever you made the first payment
+
+Thus, each offer containing a recurring payment has:
+1. A `time_unit` defining 0 (seconds), 1 (days), 2 (months).
+2. A `period`, defining how often (in `time_unit`) it has to be paid.
+3. An optional `offer_recurrence_limit` of total payments to be paid.
+4. An optional `offer_recurrence_base`:
+   * `basetime`, defining when the first period starts in seconds since 1970-01-01 UTC.
+   * The payer uses `invreq_recurrence_start` to indicate what period they are starting at.  If you don't want them to start at arbitrary periods, use `offer_absolute_expiry`.
+   * A `proportional_amount` flag: if set indicating that a payment made during the period itself will be charged proportionally to the remaining time in the period (e.g. 150 seconds into a 1500 second period gives a 10% discount).
+5. An optional `offer_recurrence_paywindow`:
+   * `seconds_before`, defining how many seconds prior to the start of the period a payment will be accepted.
+   * `seconds_after`, defining how many seconds after the start of the period a payment will be accepted.
+  If this field is missing, payment will be accepted during the prior period and the paid-for period.
+
+Note that the `offer_absolute_expiry` field already covers the case where an offer is no longer valid after January 1st 2021.
```

Comment:
I’m not sure if I understand the context this line is referencing to. It would be awesome if you can help me understand the reference here. Thanks a lot! :)

Reply:
Updated:

Note: you can create an recurring offer with a fixed start which can only be used from the first period, using `offer_absolute_expiry`.  For example, my weekly book club starts January 1st 2026, for $5 a week: by setting `offer_absolute_expiry` to January 8th 2026, I ensure nobody can join after the first week.  Now, if only I could require them to actually read this week's book!

Reply:
Looks good! (Also, wish there were a `book_reading_absolute_expiry` too 😄)